app/scripts/controllers/import.js

```diff
@@ -0,0 +1,366 @@
+                console.error(e);
+
+                return page.source;
+              }
+
+              return pageUrl.href;
+            }
+          });
+
+        var subjectUrl = '';
+
+        if (typeof subject === 'string') {
+          try {
+            subjectUrl = new URL(subject).href;
+          } catch (e) {
+            console.error('Expected valid url in import assertion subject.');
+
+            return false;
+          }
+        }
+
+        if (
+          typeof subject === 'object' &&
+          subject.toString() === '[object Object]' &&
+          subject.source !== undefined
+        ) {
+          try {
+            subjectUrl = new URL(subject.source).href;
+          } catch (e) {
+            console.error('Expected valid url in import assertion subject.');
+
+            return false;
+          }
+        }
+
+        return (sampleUrls.indexOf(subjectUrl) >= 0);
+      }
+
+      function isWcagRelated (assertionTest) {
+        function wcagIn (text) {
+          return text.indexOf('WCAG') >= 0;
+        }
+
+        // Mark the assertion.test with an WCAG id
+        // to easily match to a criterion in the auditModel when updating.
+        // It may be found top level or deep nested, finding test.wcagId is certain.
+        function setWcagIdonTest (wcagId, test) {
+          test.wcagId = wcagId;
+        }
+
+        if (
+          typeof assertionTest === 'string' &&
+          wcagIn(assertionTest)
+        ) {
+          return true;
+        }
+
+        if (
+          typeof assertionTest === 'object' &&
+          assertionTest.id !== undefined &&
+          wcagIn(assertionTest.id)
+        ) {
+          setWcagIdonTest(assertionTest.id, assertionTest);
+          return true;
+        }
+
+        if (
+          typeof assertionTest === 'object' &&
+          assertionTest.isPartOf !== undefined &&
+          typeof assertionTest.isPartOf === 'string' &&
+          wcagIn(assertionTest.isPartOf)
+        ) {
+          setWcagIdonTest(assertionTest.isPartOf, assertionTest);
+          return true;
+        }
+
+        return false;
+      }
+
+      function hasResult (_assertion) {
+        var result = _assertion.result;
+
+        function hasOutcomeValue (_result) {
+          var earlOutcome = types.EARL.OUTCOME;
+          var outcomeValues = [
+            earlOutcome.PASSED,
+            earlOutcome.FAILED,
+            earlOutcome.CANT_TELL,
+            earlOutcome.INAPPLICABLE,
+            earlOutcome.UNTESTED
+          ];
+          var outcomeClasses = [
+            earlOutcome.PASS,
+            earlOutcome.FAIL,
+            earlOutcome.CANNOT_TELL,
+            earlOutcome.NOT_APPLICABLE,
+            earlOutcome.NOT_TESTED
+          ];
+
+          if (_result.outcome === undefined) {
+            return false;
+          }
+
+          if (typeof _result.outcome === 'string') {
+            return outcomeValues.indexOf(_result.outcome) >= 0;
```

Comment:
- [ ] Move to conditional and make it return false (outcome should be listed in outcomeValues)